### PR TITLE
skip most of mark-messages-seen logic for pubchats

### PR DESCRIPTION
**Pull request by rasom**
improves (partially, for pubchats) scenario form #3336

status: ready

### CODE REVIEW

**Comment by janherich** [comment]
Unfortunately, this won't work correctly for some specific scenarios - You can't just clear all `:unviewed-messages` upon opening chat, because `:unviewed-messages` contains also ids of messages not yet (lazy) loaded, so in case you have 100 unviewed messages and you open a chat loaded from realm (where by default only first page = 20 messages are loaded) you clear all of them even when you never scrolled up, so not all unviewed messages were loaded.
Second thing, I'm pretty sure, `:unviewed-messages` set is only used for the unviewed counter, but actual statuses under the message (whether you saw the message, etc.) are dependent on the `:user-statuses` map in particular message - this is minor point, as for the public chat messages, I guess it doesn't make difference when the message has `:received` state for logged user instead of the `:seen`, but I'm not sure.

**Comment by rasom** [comment]
@janherich i just checked how it works in release, and we don't show unseen messages counter for pubchat (even if messages were not viewed) after restarting the app. Which means, i agree that this is not correct behaviour, but this is completely separate issue

File: src/status_im/chat/events.cljs [hunk 1]
```diff
@@ -224,31 +224,33 @@
...
-              (seq unseen-system-messages-ids))
-      (handlers-macro/merge-fx cofx
-                               {:db (-> (reduce (fn [new-db message-id]
-                                                  (assoc-in new-db (into messages-path [message-id :user-statuses me]) :seen))
-                                                db
-                                                (into unseen-messages-ids unseen-system-messages-ids))
-                                        (update-in [:chats chat-id :unviewed-messages] set/difference unseen-messages-ids unseen-system-messages-ids))}
-                               (persist-seen-messages chat-id (into unseen-messages-ids unseen-system-messages-ids))
-                               (send-messages-seen chat-id unseen-messages-ids)))))
+  (if (get-in db [:chats chat-id :public?])
```
Comment (cammellos) [inline src/status_im/chat/events.cljs:227]: Could you add a `public-chat?` method here https://github.com/status-im/status-react/blob/beb7cd2b84efa03d893a42e392c5550dde38369b/src/status_im/chat/models.cljs#L15 ? In the same way as https://github.com/status-im/status-react/blob/79aadec53edcf2907586c90ab0cd356c0352755e/src/status_im/chat/models.cljs#L19 .

Comment (rasom) [reply src/status_im/chat/events.cljs:227]: yes sure

**Comment by janherich** [comment]
@rasom Correct, I just checked and indeed `:unviewed` isn't working correctly for app restarts, just opening PR fixing it.

**Comment by rasom** [comment]
also if we are going to fix this `:unviewed` issue we need to show user in chat that they need to scroll in order to view unseen messages. Like this happens in slack, for instance.

**Comment by janherich** [comment]
@rasom I don't think it's really necessary as it's quite intuitive (you have to actually scroll to see messages which does not fit into the window), but sure, I agree that some additional text would be nice.

**Comment by rasom** [comment]
> as it's quite intuitive

i can't agree with this, but that's rather UX question so not for me to answer 100% :D like it's not intuitive that you need to scroll up, and you actually don't know when you scrolled far enough (unless you read each message in reverse order). 

what i saw in other messengers, for instance in telegram, it shows unseen messages starting from the oldest one. In this case definitely no need to add anything extra, because, yes, it is intuitive that you need to scroll down. 

Probably not the best place for this discussion :)

**Pull request closed without merging**